typescript-generator/index.ts

```diff
@@ -254,13 +252,13 @@ function buildRequest (type: M.Request): string {
     if (pathPropertiesNames.includes(property.name)) continue
     code += `  ${cleanPropertyName(property.name)}${property.required ? '' : '?'}: ${buildValue(property.type, openGenerics)}\n`
   }
-  if ((type.body != null) && type.body.kind === 'properties') {
+  if (type.body.kind === 'properties') {
```

Comment:
You should also check if `type.body.properties` contains something, otherwise you can generate definitions like:
```ts
export interface FooRequest {
  body?: {
  }
}
```

```suggestion
  if (type.body.kind === 'properties' && type.body.properties.length > 0) {
```

Reply:
If we enforce the existence of `body` in the spec, then requests/responses that extend a parent class with no additional properties need to have this kind of declaration. We have a few responses like this, e.g. children of `WriteResponseBase`.

And if an empty property body is used without extending a parent class, then it's an error in the spec that we can be checked in model validation, so that we don't have to check it in the code generation.

Reply:
Enforcing the existence of `body` in the response is ok, but it's not in the request.
Please add that check in the generator, we can remove it later, but for now it avoids generating ugly/useless definitions :)

Reply:
Got it and applied your suggestion. No more ugly definitions :-)